AsyncDisplayKit/TextKit/ASTextKitRenderer.mm

```diff
@@ -144,9 +162,18 @@ - (void)_calculateSize
   if (isinf(_constrainedSize.width) == NO && [_attributes.pointSizeScaleFactors count] > 0) {
     _currentScaleFactor = [[self fontSizeAdjuster] scaleFactor];
   }
-  
-  __block NSTextStorage *scaledTextStorage = nil;
+
+  // If we do not scale, do exclusion, or do custom truncation, we should just use TextKit for a fast-path.
   BOOL isScaled = [self isScaled];
+  if (isScaled == NO && self.usesCustomTruncation == NO && self.usesExclusionPaths == NO) {
```

Comment:
Is there a point at which we can lock to collect these various values for usesCustomTruncation, isScaled, etc with one lock event?  Or are all of these methods assumed to be called when locked anyway?